util/type_traits.h

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Common type traits support
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_UTIL_TYPE_TRAITS_H
+#define __SYCLCTS_UTIL_TYPE_TRAITS_H
+
+#include <type_traits>
+
+namespace {
+/**
+ * @brief Checks whether T is a floating-point sycl type
+ */
+template <typename T> struct is_cl_float_type {
+  static constexpr bool value = std::is_floating_point<T>::value ||
+                                std::is_same<cl::sycl::half, T>::value ||
+                                std::is_same<cl::sycl::cl_float, T>::value ||
+                                std::is_same<cl::sycl::cl_double, T>::value ||
+                                std::is_same<cl::sycl::cl_half, T>::value;
```

Comment:
These exist only in the OpenCL back-end.
Is the CTS validating the OpenCL back-end when running on top of OpenCL?

Reply:
Since cl_float, cl_double, cl_half are still in the CTS right now, floating point detection still takes them into consideration too. Will be removed in the future with other coverage for these types.